alpa/pipeline_parallel/primitive_def.py

```diff
@@ -24,7 +24,7 @@
 def mark_pipeline_boundary():
     """Mark the boundary of pipeline layers. We reuse pipeline_marker for this
     functionality."""
-    return pipeline_p.bind(name="boundary", mark_type="boundary")
+    return pipeline_p.bind(0, name="boundary", mark_type="boundary")
```

Comment:
If this primitive takes no input and returns no output, it will trigger a bug in jaxpr tracing. I just add a random number here to workaround the bug.

Reply:
What exactly is the bug? We have been using this boundary for a long while but we didn't to meet any bug before.

Reply:
I fixed it.